quinn-proto/src/cid_queue.rs

```diff
@@ -63,27 +64,45 @@ impl CidQueue {
         orig_offset..sequence
     }
 
-    /// Returns a new CID if any were available and a possibly-empty range preceding it to retire
-    pub fn next(&mut self) -> Option<(IssuedCid, Range<u64>)> {
-        for i in 0..Self::LEN {
+    /// Switch to next active CID if possible, return
+    /// 1) the corresponding ResetToken and 2) a possibly-empty range preceding it to retire
+    pub fn next(&mut self) -> Option<(ResetToken, Range<u64>)> {
+        for i in 1..Self::LEN {
```

Comment:
I feel like this could be written more clearly like this:

```rust
let reset_token = (1..Self::LEN).into_iter().find_map(|i| {
    let index = (self.cursor + i) % Self::LEN;
    self.buffer[index].map(|(_, token)| *token)
})?;
```

Reply:
Combined with your other comments on `fn iter_next()`, I gave a bit thought on the API format and the result is not exactly same as you suggested here. See commit `720431a`